core/src/main/java/org/sonarsource/sonarlint/core/smartnotifications/SmartNotifications.java

```diff
@@ -126,6 +129,11 @@ private void manageNotificationsForConnection(ServerApi serverApi, Map<String, S
     }
   }
 
+  private boolean shouldShowNotification(AbstractConnectionConfiguration connection, ServerNotification n) {
```

Comment:
Could we reverse this to make it `intentional | clear` 😂 
method `shouldIgnoreNotification()` and condition would be `connection.getKind() == ConnectionKind.SONARCLOUD ||
      webSocketService.hasOpenConnection() || Objects.equals(n.category(), "QUALITY_GATE")`